numpy/lib/nanfunctions.py

```diff
@@ -1292,18 +1404,29 @@ def _nanquantile_ureduce_func(a, q, axis=None, out=None, overwrite_input=False,
     return result
 
 
-def _nanquantile_1d(arr1d, q, overwrite_input=False, interpolation='linear'):
+def _nanquantile_1d(arr1d, q, weights, overwrite_input=False,
+                    interpolation='linear'):
     """
     Private function for rank 1 arrays. Compute quantile ignoring NaNs.
     See nanpercentile for parameter usage
     """
+    # remove the weights that correspond to nans, leveraing _remove_nan_1d()
+    if weights is not None:
+        wgt = weights.astype(float)
+        ind, = np.nonzero(np.isnan(arr1d))
+        if len(ind) > 0:
+            wgt[ind] = np.nan
+            weights, _ = _remove_nan_1d(wgt, overwrite_input=overwrite_input)
```

Comment:
What if the weights themselves contain `nan`? What you have here isn't consistent - the `nan`s in `wgt` only get removed if `ind` also contains `nan`.

Reply:
In `function_base.validate_weights()` I raise when the input weights contain `nan`.
My thinking is that, at this point of the code path, there shouldn't be any nan in `wgt`.